tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson} from "./SignFromJson.s.sol";
+import {DeployConfig} from "@eth-optimism-bedrock/scripts/DeployConfig.s.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {Constants, ResourceMetering} from "@eth-optimism-bedrock/src/libraries/Constants.sol";
+import {L1StandardBridge} from "@eth-optimism-bedrock/src/L1/L1StandardBridge.sol";
+import {L2OutputOracle} from "@eth-optimism-bedrock/src/L1/L2OutputOracle.sol";
+import {ProtocolVersion, ProtocolVersions} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {SuperchainConfig} from "@eth-optimism-bedrock/src/L1/SuperchainConfig.sol";
+import {OptimismPortal} from "@eth-optimism-bedrock/src/L1/OptimismPortal.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
```

Comment:
Hmm i bet its not possible to use `Deployer` because it inherits from script, seems like we need more decoupling

https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/Deployer.sol

That would save all of the logic of needing to set deploy config contract

Reply:
good idea!